Dockerfile

```diff
@@ -0,0 +1,65 @@
+FROM amazoncorretto:21 AS builder
+ARG SERVER_PORT
+ARG PG_HOST
+ARG PG_PORT
+ARG PG_DATABASE
+ARG PG_USERNAME
+ARG PG_PASSWORD
+ARG MONGO_CONNECTION_URI
+ARG MONGO_DATABASE
+ARG MAIL_USERNAME
+ARG MAIL_PASSWORD
+ARG OTP_SECRET_TOKEN
+ARG AUTH_TOKEN_PUBLIC_KEY
+ARG AUTH_TOKEN_PRIVATE_KEY
+WORKDIR /app
+COPY gradle gradle
+COPY src src
+COPY gradlew gradlew
+COPY build.gradle.kts build.gradle.kts
+COPY settings.gradle.kts settings.gradle.kts
```

Comment:
Copy everything else except src and then put `RUN ./gradlew check --dry-run` and then copy src to allow caching of the Gradle dependencies